cuML/src/knn/knn.cu

```diff
@@ -86,12 +89,9 @@ namespace ML {
 				// and have the reducer/combiner re-label the indices
 				// based on the shards they came from.
 
-				std::cout << "Adding index " << i << "total_n=" << this->total_n << std::endl;
+				if(this->verbose)
```

Comment:
Maybe a question to ponder and not necessarily a needed change right now, but I feel that the user should be able to set a single call as verbose instead of only at the constructor level i.e. what if I already fit my model, but want to debug predictions?

Reply:
You do bring up a neat idea and I believe we should discuss this in a follow-on proposal issue, perhaps post-GTC. The current use of the `verbose=True` (see [1] below for an example) in the constructor is modeled directly after sklearn, though I do see much value in, perhaps, treating the constructor value as a default and allowing any method that's called on the user API to override that default.

[1] https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html

Reply:
I put this is in a ticket and referenced this conversation.